tests/map/test-map.c

```diff
@@ -0,0 +1,108 @@
+#include <assert.h>
+#include <stdbool.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <time.h>
+
+#include "map.h"
+
+static void swap(int *x, int *y)
+{
+    int tmp = *x;
+    *x = *y;
+    *y = tmp;
+}
+
+enum { N_NODES = 10000 };
+
+/* return 0 on success; non-zero values on failure */
+static int test_map_mixed_operations()
+{
+    int failed = 0;
```

Comment:
Rename to `ret` for general usage.

Reply:
I think a have a better approach, can you check at the newest update :)